suite.go

```diff
@@ -348,12 +356,16 @@ func (suite TestSuite) executeTest(test TestSpec, extractedFields map[string]int
 	// Otherwise, deep compare response payload to expected response payload
 	var r interface{}
 	err = json.Unmarshal(body, &r)
-	if err != nil {
-		testErrors = append(testErrors, fmt.Sprintf("Error parsing json response from server: %v", err))
-		return Failed(test.Name, testErrors, time.Since(start))
-	}
-	switch r.(type) {
-	case map[string]interface{}:
+	switch {
+	case err != nil:
+		// If JSON unmarshalling fails, compare the response as a plain text string
+		expectedString, ok := expectedResponse.(string)
+		if !ok {
+			testErrors = append(testErrors, fmt.Sprintf("Expected a JSON object, but got a non-JSON response: %s", string(body)))
+		} else if string(body) != expectedString {
+			testErrors = append(testErrors, fmt.Sprintf("Expected response payload %s but got %s", expectedString, string(body)))
```

Comment:
Do we still need to return a test failure result after this line?
```go
return Failed(test.Name, testErrors, time.Since(start))
```

Reply:
It's handled at the bottom of this function: 

```
if len(testErrors) > 0 {
        // Append raw server response payload to errors for easier debugging
	testErrors = append(testErrors, fmt.Sprintf("Full response payload from server: %s", string(body)))
	return Failed(test.Name, testErrors, time.Since(start))
}
```

Reply:
Ahh ok, I missed that